src/views/components/inputs/ClearInput.tsx

```diff
@@ -41,10 +41,27 @@ type ClearInputProps = InputHTMLAttributes<HTMLInputElement> & {
 
 export const ClearInput = forwardRef<HTMLInputElement, ClearInputProps>((props, ref) => {
   const { onClear, ...inputProps } = props;
+  const [isIconShown, setIsIconShown] = useState(props.value !== '');
+
+  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (props.onChange) {
+      props.onChange(event);
+    }
+    setIsIconShown(event.target.value !== '');
+  };
+
+  const handleClear = () => {
+    onClear();
+    setIsIconShown(false);
+    if (ref && typeof ref === 'object' && ref.current) {
```

Comment:
Checking that ref is an object is probably redundant.

Reply:
Le problème si je retire cette condition j'ai une erreur ts  "Property 'current' does not exist on type..." ?
Il y a t-il une autre manière de faire pour éviter le typeof ?

Reply:
Normalement ça aurait du suffir de vérifier que ref !== null mais si ça ne suffit pas tant pi, je resolve :)